core/editor/src/tool/tools/select.rs

```diff
@@ -186,6 +222,60 @@ impl Fsm for SelectToolFsmState {
 	}
 }
 
+enum AlignDimension {
+	X,
+	Y,
+}
+
+enum AlignAggregate {
+	Min,
+	Max,
+	Average,
+}
+
+fn align_selected(document: &Document, responses: &mut VecDeque<Message>, dimension: AlignDimension, aggregate: AlignAggregate) {
+	let selected_paths: Vec<Vec<LayerId>> = document.layer_data.iter().filter_map(|(path, data)| data.selected.then(|| path.clone())).collect();
+	let n_selected = selected_paths.len();
+	if n_selected == 0 {
+		return;
+	}
```

Comment:
Opinionated style nitpick: why not just skip assigning this to a variable since it's only used here and once again down below? `selected_paths.len()` is nearly the same length as `n_selected`. Ignore this feedback if you have to make a special trip to edit this file.

Reply:
Fair enough, changed